src/neo4j/mod.rs

```diff
@@ -0,0 +1,702 @@
+            },
+            pass: Value::Env {
+                var: "NEO4J_TEST_PASS",
+                fallback: Self::DEFAULT_PASS,
+            },
+            enterprise: false,
+            plugins: Vec::new(),
+        }
+    }
+
+    /// Set the Neo4j version to use.
+    /// The value must be an existing Neo4j version tag.
+    ///
+    /// Only a subset of Semantic Versions are supported.
+    /// The version must be of the format
+    ///
+    ///    MAJOR[.MINOR[.PATCH]]
+    ///
+    ///
+    /// # Errors
+    ///
+    /// If the version is not valid according to the format.
+    pub fn with_version(
+        mut self,
+        version: impl Into<String>,
+    ) -> Result<Self, Box<dyn std::error::Error + Sync + Send + 'static>> {
+        let version: String = version.into();
+
+        let version_valid =
+            lenient_semver::parse_into::<'_, ValidateVersion>(&version).unwrap_or(false);
+        if !version_valid {
+            return Err(format!("Invalid version: {}", version).into());
+        }
+
+        self.version = Value::Value(version);
+        Ok(self)
+    }
+
+    /// Set the username to use.
+    #[must_use]
+    pub fn with_user(mut self, user: impl Into<String>) -> Self {
+        self.user = Value::Value(user.into());
+        self
+    }
+
+    /// Set the password to use.
+    #[must_use]
+    pub fn with_password(mut self, pass: impl Into<String>) -> Self {
+        self.pass = Value::Value(pass.into());
+        self
+    }
+
+    /// Do not use any authentication on the testcontainer.
+    ///
+    /// Setting this will override any prior usages of [`Self::with_user`] and
+    /// [`Self::with_password`].
+    pub fn without_authentication(mut self) -> Self {
+        self.user = Value::Unset;
+        self.pass = Value::Unset;
+        self
+    }
+
+    /// Use the enterprise edition of Neo4j.
+    ///
+    /// # Note
+    /// Please have a look at the [Neo4j Licensing page](https://neo4j.com/licensing/).
+    /// While the Neo4j Community Edition can be used for free in your projects under the GPL v3 license,
+    /// Neo4j Enterprise edition needs either a commercial, education or evaluation license.
+    pub fn with_enterprise_edition(
+        mut self,
+    ) -> Result<Self, Box<dyn std::error::Error + Sync + Send + 'static>> {
+        const ACCEPTANCE_FILE_NAME: &str = "container-license-acceptance.txt";
+
+        let version = Self::value(&self.version).expect("Version is always set");
+        let image = format!("neo4j:{}-enterprise", version);
+
+        let acceptance_file = std::env::current_dir()
+            .ok()
+            .map(|o| o.join(ACCEPTANCE_FILE_NAME));
+
+        let has_license_acceptance = acceptance_file
+            .as_deref()
+            .and_then(|o| std::fs::File::open(o).ok())
+            .into_iter()
+            .flat_map(|o| std::io::BufReader::new(o).lines())
+            .any(|o| o.map_or(false, |line| line.trim() == image));
```

Comment:
I was lifting this from the Java testcontainer [docs](https://java.testcontainers.org/modules/databases/neo4j/#choose-your-neo4j-license). It's [implemented](https://github.com/testcontainers/testcontainers-java/blob/1a3b23385f9a595576246a4c7f5dc54d244dd69d/modules/neo4j/src/main/java/org/testcontainers/containers/Neo4jContainer.java#L225) by using the [LicenseAcceptance](https://github.com/testcontainers/testcontainers-java/blob/1a3b23385f9a595576246a4c7f5dc54d244dd69d/core/src/main/java/org/testcontainers/utility/LicenseAcceptance.java#L18) utility, which seems to be from testcontainers itself, not something Neo4j specific.

I think this was also the reason to validate the version, so that users would have to use this method instead of just setting an enterprise version.

Since we removed the version validation, I can also remove this method here. I can still start an enterprise containers by using `RunnableImage` and that's good enough for us.

Reply:
Thanks for the references!
We may think of implementing something similar actually - reusable util for license acceptance.

Let's start without this check and we may create a separate PR to support something like that, If it sounds good to you

Reply:
Sure, sounds good. I already removed the method :)